### DISCUSSION TITLE
Can't export tests from insomnia to use in inso cli.

### DISCUSSION THREAD

**Discussion by PatrickGhosn** [Kong/insomnia#3958] [Help] [answered]
When I try to export the tests I have made in the test suite, I only get the option to export the normal requests and the spec, but not the unit tests. Even when I export it from the command line with `inso export spec` I get the same issue.

![image](https://user-images.githubusercontent.com/33845447/130992550-96af17f3-e073-4a21-9de4-003fd53ac260.png)

I'm not sure if I'm doing this right, of if there's another way to export the tests to use in our CI pipelines. When I run `inso run test` from my machine it detects the tests normally but there's no way to export them to the pipelines.

> inso run test
> √ Select a document or unit test suite · Indexer Test - uts_18be4f
> √ Select an environment · Indexer - env_env_845cd0
> 
> 
>   Indexer Test
>     √ Get Indexed Embeddings (1141ms)
>     √ Start Indexing Job
> 
> 
>   2 passing (1s)

**Comment by develohpanda** [comment, marked as answer]
Hi @PatrickGhosn!

I'm glad you've got the tests running locally. When run locally, Insomnia is using your Insomnia [application data directory](https://support.insomnia.rest/article/202-application-data) as a data source and reads the unit tests from there.

Currently, the only other data source available for inson is a `.insomnia` folder. For CI purposes, the intention is you enable [Git Sync](https://support.insomnia.rest/article/193-git-sync) on your Design Document within Insomnia, and sync that to a remote git repository. During that sync process, Insomnia will place all relevant data inside the `.insomnia` folder, and inso can read from that automatically when running in CI.

There is some documentation about the data source [here](https://support.insomnia.rest/article/118-cli-data-source).

---

To address your first note about only being able to select requests to export; all unit tests will be exported by default, so although you don't see an option to select unit tests like you do requests, they will be exported into the Insomnia V4 export file. Apologies for the confusion here.

As an aside, once merged, https://github.com/Kong/insomnia/pull/3055 will allow you to use an Insomnia V4 export file as _another_ data source for inso, should you choose to not use the integrated git sync, but rather manually export your document (environments, requests, unit tests, etc) into a file.

**Reply by PatrickGhosn** [reply]
Ok! Thank you for your reply. I just tried it with the app data dir and it works great. Of course the export file would make a much more cleaner solution for us but for now we can get it up and running with the `appDataDir`. :)
